client/extensions/woocommerce/state/sites/shipping-zone-locations/actions.js

```diff
@@ -20,7 +20,7 @@ export const fetchShippingZoneLocations = ( siteId, zoneId ) => ( dispatch, getS
 		return;
 	}
 
-	//0 is Rest of the World zone, skip fetching locations
+	//0 is Locations not covered by your other zones zone, skip fetching locations
```

Comment:
I think you just need to undo your changes here @jameskoster, since they are related to the API response & state tests. and @DanReyLop can you add a commit to this PR that does your reducer logic so the change is reflected on the UI side? Then I think this can move forward.